msprime/demography.py

```diff
@@ -1754,6 +1813,82 @@ def from_old_style(
                 population_map,
             )
 
+    @staticmethod
+    def from_demes(graph):
+        def get_growth_rate(epoch):
+            ret = 0
+            if epoch.size_function not in ["constant", "exponential"]:
+                raise ValueError(
+                    "msprime only supports constant exponentially changing "
+                    "population sizes"
+                )
+            if epoch.end_size != epoch.start_size:
+                ret = -math.log(epoch.start_size / epoch.end_size) / epoch.time_span
+            return ret
+
+        graph = graph.in_generations()
+
+        demography = Demography()
+        for deme in graph.demes:
+            initial_size = 0
+            growth_rate = 0
+            last_epoch = deme.epochs[-1]
+            if last_epoch.end_time == 0:
+                initial_size = last_epoch.end_size
+                growth_rate = get_growth_rate(last_epoch)
+            demography.add_population(
+                name=deme.name,
+                description=deme.description,
+                growth_rate=growth_rate,
+                initial_size=initial_size,
+                default_sampling_time=deme.end_time,
+            )
+            for epoch in deme.epochs:
+                new_initial_size = None
+                if initial_size != epoch.end_size:
+                    new_initial_size = epoch.end_size
+                new_growth_rate = None
+                alpha = get_growth_rate(epoch)
+                if growth_rate != alpha:
+                    new_growth_rate = alpha
+                if new_growth_rate is not None or new_initial_size is not None:
```

Comment:
Do you also need to check that the `epoch.end_time` is not 0, since for those demes that exist at time zero, you've already added that population's initial size and growth rate above? Maybe it's just redundant, and that's fine?